build_tools/github/get_latest_dependencies.py

```diff
@@ -0,0 +1,42 @@
+"""This script reads our requirements.txt file and removes the pinned versions"""
+
+import re
+import os
+from os.path import abspath, dirname
+
+TOP_LEVEL = abspath(dirname(dirname(dirname(__file__))))
+REQUIREMENTS = os.path.join(TOP_LEVEL, 'requirements.txt')
+BUILD_REQUIREMENTS = os.path.join(TOP_LEVEL, 'build_tools', 'build_requirements.txt')
+
+
+def find_latest_dependencies(*requirements_files):
+    """Given one or more requirements.txt files, strip off any pinned versions
+
+    Parameters
+    ----------
+    *requirements_files : str
+        One or more paths to requirements.txt files to parse
+
+    Returns
+    -------
+    requirements : list
+        List of parsed dependencies without their pinned versions
+    """
+    requirements = []
+    for requirements_file in requirements_files:
+        with open(requirements_file) as file:
+            for line in file:
+                requirement = line.strip()
+                if line.startswith('#'):
+                    continue
+                match = re.match(r'^([A-Za-z\-0-9]+)', requirement)
+                if match.group(0).lower() not in requirements:
+                    requirements.append(match.group(0).lower())
```

Comment:
I'm not super clear on this regex. But I get the purpose of the loop. FWIW (and it makes no real difference here because the size is so small and is not user-facing), set lookups are O(1) while list lookups are O(N)

Reply:
I initially had a set, but our requirements file has a specific order that I want to maintain. It's important to install Cython and Numpy before anything else, and if I use a set and cast to a list, I can't guarantee that 😕

Reply:
Also, what this regex is doing is looking for any letter characters, dashes, or numbers starting at the beginning of the string. So, it will either match anything up until our pins (`>,<,~,=`), or until the end of the line